src/dropdown/dropdown.js

```diff
@@ -170,9 +162,50 @@ angular.module('ui.bootstrap.dropdown', ['ui.bootstrap.position'])
         return;
       }
       dropdownCtrl.dropdownMenu = element;
+
+      var dropdownMenu = element,
+          dropdown = element.parent(),
+          appendToBody = dropdown.attr('dropdown-append-to-body');
+
+      scope.$watch(dropdownCtrl.isOpen, function( isOpen ) {
+        if ( isOpen ) {
+
+          // Add invisible dropdownMenu to DOM to calculate its height
+          dropdownMenu.css({
+            visibility: 'hidden',
+            display:    'block'
+          });
+
+          var windowBottom = $window.innerHeight + $window.pageYOffset,
+              dropdownOffset = $position.offset(dropdown).top,
+              dropdownHeight = $position.position(dropdown).height,
+              dropdownMenuHeight = $position.position(dropdownMenu).height,
+              dropdownMenuBottom = dropdownOffset + dropdownHeight + dropdownMenuHeight;
+
+          if ( appendToBody !== undefined ) {
```

Comment:
That is what I am confused about - further up it is set to `dropdown.attr('dropdown-append-to-body')` at the beginning of the linking function, that would presumably be set by the time the watcher is evaluated. As far as I can tell, there is no situation where it would ever be undefined.